pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1280,8 +1280,8 @@ func (c *AWSCloud) AttachDisk(diskName string, instanceName string, readOnly boo
 
 	attached := false
 	defer func() {
-		if !attached {
-			awsInstance.releaseMountDevice(disk.awsID, mountDevice)
+		if attached {
```

Comment:
Maybe change this to 'if attached || errorOccured' and set 'errorOccured' to true when 'ec2.AttachVolume()' returns an error